src/main/java/seedu/address/model/entity/CommandType.java

```diff
@@ -0,0 +1,35 @@
+package seedu.address.model.entity;
+
+/**
+ * Represents a CommandType to indicate the CommandType of a command.
+ * Guarantees: CommandType values are validated according to enum type, immutable.
+ */
+public enum CommandType {
+    P, // Participant
+    M, // Mentor
+    T, // Team
+    H; // History
+
+    // Constants
+    public static final String MESSAGE_CONSTRAINTS =
+            "CommandType should be a string of either one of the following values:\n"
+                    + "P: to indicate that a command is of type Participant\n"
+                    + "M: to indicate that a command is of type Mentor\n"
+                    + "T: to indicate that a command is of type Team\n"
+                    + "H: to indicate that a command is of type History\n";
```

Comment:
You mentioned just now you might wanna add one more CommandType here, for every other type of command? So that you no longer need to deal with null in MainWindow.java. You will be implementing this in a future update?